General Cleanup (#65)
CMakePP/CMakePPLang: docs/source/conventions.rst

Language Conventions
********************

While you can always check the CMakePPLang documentation for how to 
use a function, CMakePPLang strives to also make it as easy as possible to
"guess" how to use functions. Key to this effort is understanding 
the various conventions underlying how CMakePPLang decides on APIs. This
chapter goes into more details regarding CMakePPLang conventions.

.. note::

   CMakePPLang strives to have no exceptions to these conventions. 
   If an exception exists (and it is not documented why it is an exception) 
   assume it is a bug and please file an `issue on GitHub 
   <https://github.com/CMakePP/CMakePPLang/issues>`__.

Namespaces
==========

> CMakePPLang has a concept of variable scope, but does not have a 

Reviewer: ```suggestion
Both CMake and CMakePP have a concept of variable scope, but neither have a 
```
